frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx

```diff
@@ -71,15 +73,53 @@ const TopologyFilterBar: React.FC<TopologyFilterBarProps> = ({
   const kialiLink = getNamespaceDashboardKialiLink(consoleLinks, namespace);
   const queryParams = useQueryParams();
   const searchQuery = queryParams.get('searchQuery') || '';
+  const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(false);
 
   const onTextFilterChange = (text) => {
     const query = text?.trim();
     onSearchChange(query);
   };
 
+  React.useEffect(() => {
+    const onKeyDown = (e: KeyboardEvent) => {
+      e.preventDefault();
+      const { nodeName } = e.target as Element;
+      if (nodeName === 'INPUT' || nodeName === 'TEXTAREA') {
+        return;
+      }
+      switch (e.code) {
+        case 'KeyF':
+          e.ctrlKey && setIsQuickSearchActive(true);
+          break;
+        case 'Escape':
+          setIsQuickSearchActive(false);
+          break;
+        default:
+          break;
+      }
+    };
+
+    window.addEventListener('keydown', onKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', onKeyDown);
+    };
+  });
+
   return (
     <Toolbar className="co-namespace-bar odc-topology-filter-bar">
       <ToolbarContent>
+        <ToolbarItem>
+          <Tooltip position="right" content="Add to project">
+            <Button
+              className="odc-topology-filter-bar__quick-search"
+              variant="plain"
+              isActive={isQuickSearchActive}
+            >
+              <QuickSearchIcon />
+            </Button>
+          </Tooltip>
```

Comment:
This should be extracted into a separate component so that we're not putting too much logic into the TopologyFilterBar component.

Reply:
Makes sense.